crates/deltalake-core/src/table/builder.rs

```diff
@@ -266,11 +279,22 @@ impl DeltaTableBuilder {
 
     /// Build a delta storage backend for the given config
     pub fn build_storage(self) -> DeltaResult<LogStoreRef> {
-        config::configure_log_store(
-            &self.options.table_uri,
-            self.storage_options(),
-            self.options.storage_backend,
-        )
+        debug!("build_storage() with {}", &self.options.table_uri);
+        let location = Url::parse(&self.options.table_uri).map_err(|_| {
+            DeltaTableError::NotATable(format!(
+                "Could not turn {} into a URL",
+                self.options.table_uri
+            ))
```

Comment:
seems like in practice we should never hit this, if the builder was crated via a rout that ensure URIs are fine ... since we do have to dela with relative paths and the likes, how about we make the field on `DeltaLoadOptios` a URL, and avoid throwing here?